### DISCUSSION TITLE
query problem

### DISCUSSION THREAD

**Discussion by maxile1** [vuejs/router#2306] [Help and Questions] [answered]
When i use this.$router.replace({path:'/pageA',query:{showFooter:false}}) in hash routing. When i get showFooter in pageA, the type of showFooter is usually string,but sometimes showFooter is boolean. I want to know why?

**Comment by maxile1** [comment]
It's always a string in my understanding

**Reply by Deckluhm** [reply, marked as answer]
It can either be a `string`, `number`, `null` or `undefined` (source: https://github.com/vuejs/router/blob/main/packages/router/src/query.ts#L23).

> but sometimes showFooter is boolean

Where do you see that?
